frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -151,6 +155,49 @@ export const insightLogic = kea<insightLogicType>({
                 startQuery: (state, { queryId }) => ({ ...state, [queryId]: new Date().getTime() }),
             },
         ],
+        lastInsightModeSource: [
+            null as InsightEventSource | null,
+            {
+                setInsightMode: (_, { source }) => source,
+            },
+        ],
+    },
+    loaders: ({ values, actions }) => ({
+        insight: {
```

Comment:
I'm a bit worried this is adding more complexity to all these insight-related logics. I thought we already had a reducer somewhere to store the current insight object?

Reply:
I agree but I also think it makes sense in there because keeping track of which insight we're currently on does seem like an `insightLogic` concern. However the editing functionality can probably moved out. Also I couldn't find that reducer that stored current insight object when I first looked, do you know where it might be?

Reply:
Got it! Let's clarify then where this is used. Particularly worth noting that when viewing a dashboard or on specific insights (where you use insight-specific logic) you don't use this reducer. Can't find it so I may have dreamt it.